Admin.js

```diff
@@ -114,561 +136,771 @@ var Admin = {
-            collection.children().each(function() {
-                var matches = highestCounterRegexp.exec(jQuery('[id^="sonata-ba-field-container"]', this).attr('id'));
-                if (matches && matches[1] && matches[1] > counter) {
-                    counter = parseInt(matches[1], 10);
-                }
+        const highestCounterRegexp = new RegExp("_([0-9]+)[^0-9]*$");
+        jQuery(subject).find("[data-prototype]").
+            each(function () {
+
+                const collection = jQuery(this);
+                let counter = 0;
+                collection.children().each(function () {
+
+                    const matches = highestCounterRegexp.exec(jQuery("[id^=\"sonata-ba-field-container\"]", this).attr("id"));
+                    if (matches && matches[1] && matches[1] > counter) {
+
+                        counter = parseInt(matches[1], 10);
+
+                    }
+
+                });
+                Admin.lolCounters[collection.attr("id")] = counter;
+
             });
-            Admin.lolCounters[collection.attr('id')] = counter;
-        });
+
     },
 
-    setup_collection_buttons: function(subject) {
+    "setup_collection_buttons" (subject) {
+
+        jQuery(subject).on("click", ".sonata-collection-add", function (event) {
 
-        jQuery(subject).on('click', '.sonata-collection-add', function(event) {
             Admin.stopEvent(event);
 
-            var container = jQuery(this).closest('[data-prototype]');
-            var counter = ++Admin.lolCounters[container.attr('id')];
-            var proto = container.attr('data-prototype');
-            var protoName = container.attr('data-prototype-name') || '__name__';
+            const container = jQuery(this).closest("[data-prototype]");
+            const counter = ++Admin.lolCounters[container.attr("id")];
+            let proto = container.attr("data-prototype");
+            const protoName = container.attr("data-prototype-name") || "__name__";
             // Set field id
-            var idRegexp = new RegExp(container.attr('id')+'_'+protoName,'g');
-            proto = proto.replace(idRegexp, container.attr('id')+'_'+counter);
+            const idRegexp = new RegExp(`${container.attr("id")}_${protoName}`, "g");
+            proto = proto.replace(idRegexp, `${container.attr("id")}_${counter}`);
 
             // Set field name
-            var parts = container.attr('id').split('_');
-            var nameRegexp = new RegExp(parts[parts.length-1]+'\\]\\['+protoName,'g');
-            proto = proto.replace(nameRegexp, parts[parts.length-1]+']['+counter);
-            jQuery(proto)
-                .insertBefore(jQuery(this).parent())
-                .trigger('sonata-admin-append-form-element')
-            ;
-
-            jQuery(this).trigger('sonata-collection-item-added');
+            const parts = container.attr("id").split("_");
+            const nameRegexp = new RegExp(`${parts[parts.length - 1]}\\]\\[${protoName}`, "g");
```

Comment:
- eslint: Line 446 exceeds the maximum line length of 80.
- eslint: Combine this with the previous 'const' statement.
- eslint: No magic number: 1.